dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -98,62 +125,108 @@ private void initializeActiveVanityURLsCacheBySiteAndLanguage(String siteId,
+                        this.getFallback(uri, siteId, languageId, this.systemUser);
+            }
+
+            isVanityURL =
+                    UtilMethods.isSet(cachedVanityUrl)
+                            && !VanityUrlAPI.CACHE_404_VANITY_URL
+                                .equals(cachedVanityUrl.getVanityUrlId());
+
+            // Still support legacy cmsHomePage
+            if (URL_SUFFIX.equals(uri) && !isVanityURL) {
+
+                cachedVanityUrl =
+                        this.getLiveCachedVanityUrl
+                                (LEGACY_CMS_HOME_PAGE, host, languageId, this.systemUser);
+                isVanityURL = UtilMethods.isSet(cachedVanityUrl)
+                        && !VanityUrlAPI.CACHE_404_VANITY_URL
+                            .equals(cachedVanityUrl.getVanityUrlId());
+            }
+        }
+
+        return isVanityURL;
+    } // isVanityUrl
 
     @CloseDB
     @Override
-    public VanityUrl getVanityUrlFromContentlet(final Contentlet con) {
-        if (con != null) {
+    public VanityUrl getVanityUrlFromContentlet(final Contentlet contentlet) {
+
+        if (contentlet != null) {
             try {
-                if (!con.isVanityUrl()) {
+                if (!contentlet.isVanityUrl()) {
                     throw new DotStateException(
-                            "Contentlet : " + con.getInode() + " is not a Vanity Url");
+                            "Contentlet : " + contentlet.getInode() + " is not a Vanity Url");
                 }
             } catch (DotDataException | DotSecurityException e) {
                 throw new DotStateException(
-                        "Contentlet : " + con.getInode() + " is not a Vanity Url", e);
+                        "Contentlet : " + contentlet.getInode() + " is not a Vanity Url", e);
             }
         } else {
             throw new DotStateException("Contentlet is null");
         }
 
-        DefaultVanityUrl vanityUrl = new DefaultVanityUrl();
-        vanityUrl.setContentTypeId(con.getContentTypeId());
+        final DefaultVanityUrl vanityUrl = new DefaultVanityUrl();
+        vanityUrl.setContentTypeId(contentlet.getContentTypeId());
         try {
-            contentletAPI.copyProperties(vanityUrl, con.getMap());
+            this.contentletAPI.copyProperties(vanityUrl, contentlet.getMap());
         } catch (Exception e) {
             throw new DotStateException("Vanity Url Copy Failed", e);
         }
-        vanityUrl.setHost(con.getHost());
-        vanityUrl.setFolder(con.getFolder());
+        vanityUrl.setHost(contentlet.getHost());
+        vanityUrl.setFolder(contentlet.getFolder());
 
         return vanityUrl;
-    }
+    } // getVanityUrlFromContentlet.
 
     /**
      * Add the Vanity URL to the vanityURLCache
      *
      * @param vanityUrl The vanityurl URL object
      */
-    private void addToVanityURLCache(VanityUrl vanityUrl) {
+    private void addToVanityURLCache(final VanityUrl vanityUrl) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "addToVanityURLCache" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)